hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/placement/algorithms/SCMContainerPlacementRackScatter.java

```diff
@@ -99,11 +98,15 @@ public SCMContainerPlacementRackScatter(final NodeManager nodeManager,
       throws SCMException {
     Preconditions.checkArgument(nodesRequired > 0);
     metrics.incrDatanodeRequestCount(nodesRequired);
-    int datanodeCount = networkTopology.getNumOfLeafNode(NetConstants.ROOT);
     int excludedNodesCount = excludedNodes == null ? 0 : excludedNodes.size();
-    if (datanodeCount < nodesRequired + excludedNodesCount) {
+    List<Node> totalNodes = networkTopology.getNodes(
+        networkTopology.getMaxLevel());
+    if (excludedNodes != null) {
+      totalNodes.removeAll(excludedNodes);
+    }
+    if (totalNodes.size() < nodesRequired) {
       throw new SCMException("No enough datanodes to choose. " +
-          "TotalNode = " + datanodeCount +
+          "TotalNode = " + totalNodes.size() +
           " RequiredNode = " + nodesRequired +
           " ExcludedNode = " + excludedNodesCount, null);
```

Comment:
How about add a new variable named totalNodeAfterExclude or availableNode?
If SCM has 10 nodes and user excluded 2 nodes, the error message would show the total node be 8, not equal to the current number SCM recognize.

Reply:
@symious Ah, that's a good point, thanks~ 
I've updated a new version.